lib/cocoapods/generator/xcconfig/xcconfig_helper.rb

```diff
@@ -52,6 +52,26 @@ def self.default_ld_flags(target, includes_static_libraries = false)
         def self.add_settings_for_file_accessors_of_target(target, xcconfig)
           target.file_accessors.each do |file_accessor|
             XCConfigHelper.add_spec_build_settings_to_xcconfig(file_accessor.spec_consumer, xcconfig)
+          end
+          XCConfigHelper.add_vendored_dependency_build_settings(target, xcconfig)
+        end
+
+        # Adds build settings for vendored frameworks and libraries.
+        #
+        # @param [PodTarget] target
+        #        The pod target, which holds the list of +Spec::FileAccessor+.
+        #
+        # @param [Xcodeproj::Config] xcconfig
+        #        The xcconfig to edit.
+        #
+        def self.add_vendored_dependency_build_settings(target, xcconfig)
+          if target.host_requires_frameworks?
+            target.dependent_targets.each do |dependent_target|
+              XCConfigHelper.add_vendored_dependency_build_settings(dependent_target, xcconfig)
```

Comment:
I don't think we need to recurse here. Dependent targets includes already transitive dependencies.